### PR TITLE
For review only, not finished : split guice modules into smaller ones (JAMES-1902)

### PR DESCRIPTION
**Pull request by mbaechler**

### CODE REVIEW

File: server/container/guice/cassandra-guice/src/main/java/org/apache/james/modules/mailbox/CassandraMailboxModule.java [hunk 1]
```diff
@@ -22,6 +22,7 @@
 
 import org.apache.james.adapter.mailbox.store.UserRepositoryAuthenticator;
 import org.apache.james.backends.cassandra.components.CassandraModule;
+import org.apache.james.mailbox.AttachmentManager;
```
Comment (chibenwa) [inline server/container/guice/cassandra-guice/src/main/java/org/apache/james/modules/mailbox/CassandraMailboxModule.java]: What about this import in Memory implementation ?

File: server/container/guice/pom.xml [hunk 2]
```diff
@@ -79,6 +79,11 @@
                 <artifactId>james-server-guice-smtp</artifactId>
                 <version>3.0.0-beta6-SNAPSHOT</version>
             </dependency>
+            <dependency>
+                <groupId>${project.groupId}</groupId>
+                <artifactId>james-server-guice-webadmin</artifactId>
+                <version>3.0.0-beta6-SNAPSHOT</version>
```
Comment (chibenwa) [inline server/container/guice/pom.xml]: project.version

File: server/container/guice/pom.xml [hunk 3]
```diff
@@ -46,6 +46,11 @@
             </dependency>
             <dependency>
                 <groupId>${project.groupId}</groupId>
+                <artifactId>james-server-guice-jmap</artifactId>
+                <version>3.0.0-beta6-SNAPSHOT</version>
```
Comment (chibenwa) [inline server/container/guice/pom.xml]: project.version

File: server/container/guice/pom.xml [hunk 4]
```diff
@@ -46,6 +46,11 @@
...
+            </dependency>
+            <dependency>
+                <groupId>${project.groupId}</groupId>
                 <artifactId>james-server-guice-lmtp</artifactId>
                 <version>3.0.0-beta6-SNAPSHOT</version>
```
Comment (chibenwa) [inline server/container/guice/pom.xml]: project.version

File: server/container/guice/pom.xml [hunk 5]
```diff
@@ -56,6 +56,11 @@
             </dependency>
             <dependency>
                 <groupId>${project.groupId}</groupId>
+                <artifactId>james-server-guice-mailet</artifactId>
+                <version>3.0.0-beta6-SNAPSHOT</version>
```
Comment (chibenwa) [inline server/container/guice/pom.xml]: project.version

File: server/container/guice/pom.xml [hunk 6]
```diff
@@ -56,6 +56,11 @@
             </dependency>
             <dependency>
                 <groupId>${project.groupId}</groupId>
+                <artifactId>james-server-guice-managedsieve</artifactId>
+                <version>3.0.0-beta6-SNAPSHOT</version>
```
Comment (chibenwa) [inline server/container/guice/pom.xml]: project.version

File: server/container/guice/pom.xml [hunk 7]
```diff
@@ -54,6 +54,11 @@
                 <artifactId>james-server-guice-pop</artifactId>
                 <version>3.0.0-beta6-SNAPSHOT</version>
             </dependency>
+            <dependency>
+                <groupId>${project.groupId}</groupId>
+                <artifactId>james-server-guice-smtp</artifactId>
+                <version>3.0.0-beta6-SNAPSHOT</version>
```
Comment (chibenwa) [inline server/container/guice/pom.xml]: project.version

File: server/container/guice/pom.xml [hunk 8]
```diff
@@ -49,6 +49,11 @@
                 <artifactId>james-server-guice-mailbox</artifactId>
                 <version>3.0.0-beta6-SNAPSHOT</version>
             </dependency>
+            <dependency>
+                <groupId>${project.groupId}</groupId>
+                <artifactId>james-server-guice-pop</artifactId>
+                <version>3.0.0-beta6-SNAPSHOT</version>
```
Comment (chibenwa) [inline server/container/guice/pom.xml]: project.version

File: server/container/guice/pom.xml [hunk 9]
```diff
@@ -32,12 +32,35 @@
...
+        <dependencies>
+            <dependency>
+                <groupId>${project.groupId}</groupId>
+                <artifactId>james-server-guice-configuration</artifactId>
+                <version>3.0.0-beta6-SNAPSHOT</version>
+            </dependency>
+            <dependency>
+                <groupId>${project.groupId}</groupId>
+                <artifactId>james-server-guice-imap</artifactId>
+                <version>3.0.0-beta6-SNAPSHOT</version>
```
Comment (chibenwa) [inline server/container/guice/pom.xml]: project.version

File: server/container/guice/pom.xml [hunk 10]
```diff
@@ -32,12 +32,35 @@
...
+            </dependency>
+            <dependency>
+                <groupId>${project.groupId}</groupId>
+                <artifactId>james-server-guice-mailbox</artifactId>
+                <version>3.0.0-beta6-SNAPSHOT</version>
```
Comment (chibenwa) [inline server/container/guice/pom.xml]: project.version

File: server/container/guice/guice-common/src/test/java/org/apache/james/AbstractJmapJamesServerTest.java [hunk 11]
```diff
@@ -60,12 +61,11 @@ public void setup() throws Exception {
         		.setContentType(ContentType.JSON)
         		.setAccept(ContentType.JSON)
         		.setConfig(newConfig().encoderConfig(encoderConfig().defaultContentCharset(Charsets.UTF_8)))
-        		.setPort(server.getJmapProbe()
-                    .getJmapPort())
+        		.setPort(server.getProbe(JmapGuiceProbe.class).getJmapPort())
```
Comment (chibenwa) [inline server/container/guice/guice-common/src/test/java/org/apache/james/AbstractJmapJamesServerTest.java]: Yep, it's more simple, I definitly upvote that ;-)

**Comment by chibenwa** [comment]
Not yet a true orthogonal architeture, right ?

We should have as well modules from data and mailbox + combine all of that for generating products. Is that your plan ?

**Comment by mbaechler** [comment]
First, thank you for reviewing !
My plan for this step is to be able to compose servers with various protocols (smtp only, imap+smtp, etc)
I agree that we will need much more modules to have something really composable (smtp only includes at least splitting data/mailbox stuff) but always driven by some actual needs.
First thing first, let's try to have that task merged !

**Comment by chibenwa** [comment]
Updated PR have been merged yesterday

**Pull request closed without merging**